http-handler/src/main/java/com/flipkart/poseidon/handlers/http/impl/HttpConnectionPool.java

```diff
@@ -231,6 +248,16 @@ public HttpResponse doPOST(String uri, byte[] data, Map<String, String> requestH
         setRequestHeaders(request, requestHeaders);
         return execute(request);
     }
+    
+    /**
+     * Method for executing HTTP POST request with form params
+     */
+    public HttpResponse doPOST(String uri, List<NameValuePair> formParams , Map<String, String> requestHeaders) throws Exception {
+        HttpPost request = new HttpPost(constructUrl(uri));
+        request.setEntity(new UrlEncodedFormEntity(formParams));
```

Comment:
For completeness, we need to support compressing form params as well. Something like
if (this.requestGzipEnabled) {
                request.addHeader(CONTENT_ENCODING, COMPRESSION_TYPE);
                request.setEntity(new GzipCompressingEntity(new UrlEncodedFormEntity(formParams)));
            } else {
                request.setEntity(new UrlEncodedFormEntity(formParams));
}

UrlEncodedFormEntity changes Content-Type header (to "application/x-www-form-urlencoded") and requestGzipEnabled controls Content-Encoding header.